services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowPriorityRerouteSpec.groovy

```diff
@@ -0,0 +1,65 @@
+package org.openkilda.functionaltests.spec.flows
+
+
+import static org.openkilda.testing.Constants.WAIT_OFFSET
+
+import org.openkilda.functionaltests.BaseSpecification
+import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.messaging.info.event.IslChangeType
+import org.openkilda.messaging.info.event.PathNode
+import org.openkilda.messaging.payload.flow.FlowPayload
+import org.openkilda.testing.model.topology.TopologyDefinition.Switch
+
+import java.util.concurrent.TimeUnit
+
+class FlowPriorityRerouteSpec extends BaseSpecification {
+    def "System is able to reroute flow in the correct order based on the priority field"() {
+        given: "3 flows on the same path, with alt paths available"
+        def switches = topology.activeSwitches
+        List<List<PathNode>> allPaths = []
+        def (Switch srcSwitch, Switch dstSwitch) = [switches, switches].combinations()
+                .findAll { src, dst -> src != dst }.unique { it.sort() }.find { Switch src, Switch dst ->
+            allPaths = database.getPaths(src.dpId, dst.dpId)*.path
+            allPaths.size() > 2
+        }
+        List<FlowPayload> flows = []
+
+        def newPriority = 100
+
+        3.times {
+            def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+            flow.maximumBandwidth = 10000
+            newPriority += 100
+            flow.priority = newPriority
+            flowHelper.addFlow(flow)
+            flows << flow
+        }
+        def currentPath = pathHelper.convert(northbound.getFlowPath(flows[0].id))
+        //ensure all flows are on the same path
+        assert pathHelper.convert(northbound.getFlowPath(flows[1].id)) == currentPath
+        assert pathHelper.convert(northbound.getFlowPath(flows[2].id)) == currentPath
+
+        and: "Alternative paths"
+        def newPath = allPaths.find { it != currentPath }
+        allPaths.findAll { it != newPath }.each { pathHelper.makePathMorePreferable(newPath, it) }
+
+        when: "Init simultaneous reroute for all flows by bringing current path's ISL down"
```

Comment:
In order to perform reroute you can do either
- Change costs so that there is a better path. Init manual reroute
or
- Just drop ISL so that flow finds a new path
You did both and I don't really understand why

Reply:
just in case
I'll leave the portDown only